tests/test_llmblock.py

```diff
@@ -86,3 +89,19 @@ def test_model_prompt_none(self, mock_load_config):
             "FOO pear\nintroduction\nprinciples\nexamples\ngeneration BAR",
             "model_prompt should be a non-empty string when set to None",
         )
+
+    @patch("src.instructlab.sdg.block.Block._load_config")
+    def test_max_num_tokens_override(self, mock_load_config):
+        mock_load_config.return_value = self.config_return_value
+        # Ensure that if a custom model_prompt is specified, it is used correctly
+        block = LLMBlock(
+            ctx=self.mock_ctx,
+            pipe=self.mock_pipe,
+            block_name="gen_knowledge",
+            max_num_token_override=512,
+            config_path="",
+            output_cols=[],
+            model_prompt="",
```

Comment:
We're missing one thing here to actually verify this is overriding the config - another arg of: 
```python
            gen_kwargs={"max_tokens": 2048},
```

That will simulate that we read a config file specifying 2048 as the max tokens (like our simple/full configs do) but that we're actually overriding that to 512 tokens via the new override parameter. Otherwise, the test is only ensuring we don't use the `DEFAULT_MAX_NUM_TOKENS` value but isn't ensuring we don't use the pipeline config file's values, which is what we want to overwrite.